lib/ecosystems/enrich_cyclonedx.go

```diff
@@ -69,6 +70,9 @@ func enrichExternalReference(comp *cdx.Component, url *string, refType cdx.Exter
 	if url == nil {
 		return
 	}
+	if _, err := gourl.Parse(*url); err != nil {
```

Comment:
nit: can we use the std import name `url` and rename this argument instead to `ref`?